railties/lib/rails/engine.rb

```diff
@@ -618,6 +618,13 @@ def load_seed
       end
     end
 
+    initializer :add_mailer_preview_paths do
+      previews = paths["test/mailers/previews"].existent
```

Comment:
The contents of `config.action_mailer.preview_paths` are added to `ActiveSupport::Dependencies.autoload_paths`, I think we should do the same with these paths too.

It's not necessary for the mailer previews to work, but it will make a naming conflict between a preview and another autoloaded constant more apparent, and I suspect the way previews are currently eager loaded ([`require_dependency` with absolute paths](https://github.com/rails/rails/blob/db543ba728a9bd38a08f62ebb93ec9eca802a987/actionmailer/lib/action_mailer/preview.rb#L123)) won't work once the classic autoloader is deprecated.

Reply:
Updated `paths.add "test/mailers/previews", eager_load: true` to serve this purpose. This should work, I believe?

Reply:
Yep, that's even better 👍 

I wonder if the railtie should add its preview paths to `paths["test/mailers/previews"]` instead of directly manipulating `ActiveSupport::Dependencies.autoload_paths`?

```patch
diff --git a/actionmailer/lib/action_mailer/railtie.rb b/actionmailer/lib/action_mailer/railtie.rb
index 6bae8c825e..2bd5342fca 100644
--- a/actionmailer/lib/action_mailer/railtie.rb
+++ b/actionmailer/lib/action_mailer/railtie.rb
@@ -55,12 +55,10 @@ class Railtie < Rails::Railtie # :nodoc:
       end
     end

-    initializer "action_mailer.set_autoload_paths" do |app|
+    initializer "action_mailer.set_autoload_paths", before: :set_autoload_paths do |app|
       options = app.config.action_mailer

-      if options.show_previews
-        ActiveSupport::Dependencies.autoload_paths.concat(options.preview_paths)
-      end
+      app.config.paths["test/mailers/previews"].concat(options.preview_paths)
     end

     initializer "action_mailer.compile_config_methods" do
```